x-pack/plugins/lens/public/shared_components/drag_drop_bucket/buckets.tsx

```diff
@@ -5,162 +5,110 @@
+}: Assign<
+  Omit<BucketContainerProps, 'draggableProvided'>,
+  {
+    id: string;
+    children: React.ReactNode;
+    isInsidePanel?: boolean;
+    Container?: React.FunctionComponent<BucketContainerProps>;
+  }
+>) => {
+  const { euiTheme } = useEuiTheme();
+
   return (
     <EuiDraggable
-      style={{ marginBottom: 4 }}
-      spacing="none"
-      index={idx}
       draggableId={id}
+      customDragHandle={true}
+      index={bucketContainerProps.idx}
+      isDragDisabled={bucketContainerProps.isNotDraggable}
+      style={!isInsidePanel ? { marginBottom: euiTheme.size.xs } : {}}
+      spacing="none"
       disableInteractiveElementBlocking
     >
-      {(provided) => <BucketContainer {...bucketContainerProps}>{children}</BucketContainer>}
+      {(provided, state) => (
+        <Container
+          draggableProvided={provided}
+          isDragging={state?.isDragging ?? false}
+          {...bucketContainerProps}
+        >
+          {children}
+        </Container>
+      )}
     </EuiDraggable>
   );
 };
 
-interface DraggableLocation {
-  droppableId: string;
-  index: number;
-}
-
-export const DragDropBuckets = ({
+export function DragDropBuckets<T = unknown>({
   items,
   onDragStart,
   onDragEnd,
   droppableId,
   children,
-  className,
+  bgColor,
 }: {
-  items: any; // eslint-disable-line @typescript-eslint/no-explicit-any
-  onDragStart: () => void;
-  onDragEnd: (items: any) => void; // eslint-disable-line @typescript-eslint/no-explicit-any
+  items: T[];
   droppableId: string;
   children: React.ReactElement[];
-  className?: string;
-}) => {
-  const handleDragEnd = ({
-    source,
-    destination,
-  }: {
-    source?: DraggableLocation;
-    destination?: DraggableLocation;
-  }) => {
-    if (source && destination) {
-      const newItems = euiDragDropReorder(items, source.index, destination.index);
-      onDragEnd(newItems);
-    }
-  };
+  onDragStart?: () => void;
+  onDragEnd?: (items: T[]) => void;
+  bgColor?: EuiPanelProps['color'];
+}) {
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleDragEnd: DragDropContextProps['onDragEnd'] = useCallback(
+    ({ source, destination }) => {
+      setIsDragging(false);
+      if (source && destination) {
+        onDragEnd?.(euiDragDropReorder(items, source.index, destination.index));
+      }
+    },
+    [items, onDragEnd]
+  );
+
+  const handleDragStart: DragDropContextProps['onDragStart'] = useCallback(() => {
+    setIsDragging(true);
+    onDragStart?.();
+  }, [onDragStart]);
+
   return (
-    <EuiDragDropContext onDragEnd={handleDragEnd} onDragStart={onDragStart}>
-      <EuiDroppable droppableId={droppableId} spacing="none" className={className}>
-        {children}
-      </EuiDroppable>
+    <EuiDragDropContext onDragEnd={handleDragEnd} onDragStart={handleDragStart}>
+      <EuiPanel
+        paddingSize="none"
+        color={isDragging ? 'success' : bgColor}
+        style={{ overflow: 'hidden' }}
```

Comment:
With the changes to where the background color is being applied, I don't believe this `style` prop is needed anymore.

```suggestion
```